common4j/src/main/com/microsoft/identity/common/java/cache/SharedPreferencesAccountCredentialCacheWithMemoryCache.java

```diff
@@ -736,4 +740,26 @@ private static boolean isCredential(@NonNull String cacheKey) {
         return isCredential;
     }
 
+    /**
+     * Tells if access tokens with SHA-1 app identifiers have been cleared yet.
+     * @return true if cleared; false otherwise.
+     */
+    @Override
+    protected boolean isSha1Cleared() {
+        synchronized (mCacheLock) {
+            waitForInitialLoad();
```

Comment:
These don't actually need to be guarded in this way -- the mCacheLock is protecting specifically the 
mCachedAccountRecordsWithKeys and mCachedCredentialsWithKeys structures.  It's safe to go touch this mSharedPreferencesFileManager without them.  I think this guarding difference was probably the cause for this becoming an abstract method, so it can probable have a single implementation back in AbstractAccount* now as well.

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/commit/4b24755cb9168aa44753751b7b993bf65e2f585f